posit-dev/positron
# Data Explorer: UI treatment for copy-and-pasting from data grid and wire up with Python implementation, context menu improvements

### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any GitHub issues that are related.

This PR enables copying data from the Positron Data Explorer to the clipboard.

It also contains some other work:

- `showContextMenu` (`ContextMenuEntry`, `ContextMenuItem`, and `ContextMenuSeparator`) has been renamed `showCustomContextMenu` (and `CustomContextMenuEntry`, `CustomContextMenuItem`, and `CustomContextMenuSeparator`).
- Some work has been done on showing column, row, and cell context menus in the Positron Data Explorer.
- Addition of `PositronDataExplorerFocused` context key, which will be true when keyboard focus is inside a Positron Data Explorer.
- Positron Data Explorer actions have been updated. The placeholder action has been removed. A new action, `workbench.action.positronDataExplorer.copy`, has been added.
- `positronDataExplorerEditor.contribution.ts` has been updated to use `registerWorkbenchContribution2`.

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

You should be able to copy data out of a Positron Data Explorer using the ⌘ C (macOS) or Ctrl + C (Windows or Linux). Here's a movie:


https://github.com/posit-dev/positron/assets/853239/f6b6eaa2-c55f-4707-874f-2fe706a5ee5a
